pandas/io/formats/format.py

```diff
@@ -1600,21 +1601,24 @@ def format_percentiles(
         return [i + "%" for i in out]
 
     unique_pcts = np.unique(percentiles)
-    to_begin = unique_pcts[0] if unique_pcts[0] > 0 else None
-    to_end = 100 - unique_pcts[-1] if unique_pcts[-1] < 100 else None
-
-    # Least precision that keeps percentiles unique after rounding
-    prec = -np.floor(
-        np.log10(np.min(np.ediff1d(unique_pcts, to_begin=to_begin, to_end=to_end)))
-    ).astype(int)
-    prec = max(1, prec)
+    prec = get_precision(unique_pcts)
     out = np.empty_like(percentiles, dtype=object)
     out[int_idx] = percentiles[int_idx].round().astype(int).astype(str)
 
     out[~int_idx] = percentiles[~int_idx].round(prec).astype(str)
     return [i + "%" for i in out]
 
 
+def get_precision(array: (np.ndarray | Sequence[float])) -> int:
+    to_begin = array[0] if array[0] > 0 else None
+    to_end = 100 - array[-1] if array[-1] < 100 else None
+    diff = np.ediff1d(array, to_begin=to_begin, to_end=to_end)
+    diff = abs(diff)
```

Comment:
Why is `abs` needed here when it wasn't used before above?

Reply:
pandas/tests/io/formats/test_format.py::TestFormatPercentiles::test_format_percentiles one of this test case is failing because  

`np.min(np.ediff1d(unique_pcts, to_begin=to_begin, to_end=to_end))`

the above statement is returning negative number. 

So that np.log10 is raising RuntimeWarning

`RuntimeWarning: invalid value encountered in log10
` 

![image](https://github.com/pandas-dev/pandas/assets/25401444/20d0c565-a8c7-438d-ae06-8253df7b764e)


that's why I added `abs`

Reply:
OK thanks for the explaination